app/controllers/comments_controller.rb

```diff
@@ -9,8 +9,12 @@ def index; end
 
   def create
     @project = Project.find(params[:project_id])
-    @comment = @project.comments.create(comment_params.merge(user_id: current_user.id, comment_owner_id: params[:project_id]))
-    redirect_to project_path(@project)
+    @comment = @project.comments.new(comment_params.merge(user_id: current_user.id, comment_owner_id: params[:project_id]))
+    if @comment.save
+      redirect_to project_path(@project)
+    else
+      redirect_to project_path(@project), alert: @comment.errors.full_messages
```

Comment:
Will merge for now, because deadline.
For the future, do NOT use flash to render validation errors, this should be done inside model for, in this case inside comment form on projects page.
errors would be shown with render, not (flash + redirect)